## Question - 1
ans - 

In [1]:
import pandas as pd

df = pd.DataFrame({'time_spent':[3,2,4,2,3],
                  'scores':[90,80,95,80,90]})

df

time_spent  scores
0           3      90
1           2      80
2           4      95
3           2      80
4           3      90

In [2]:
df.corr(method = 'pearson')

time_spent    scores
time_spent    1.000000  0.979958
scores        0.979958  1.000000

interpretation :-

1. Time Spent and Scores:


* .Correlation: 0.979958

* .The correlation of approximately 0.98 indicates a very strong positive correlation between "time_spent" and "scores."

* .This means that there is a close, positive linear relationship between the amount of time spent on a task or activity and the corresponding scores. When time spent increases, scores tend to increase as well, and when time spent decreases, scores tend to decrease.

## Question -2
ans - 

In [3]:
import pandas as pd
from scipy.stats import spearmanr

df = pd.DataFrame({'sleep' : [7, 6, 8, 5, 6, 7, 6, 8, 5, 7],
                   
                 'job_satisfaction' : [8, 5, 9, 4, 6, 8, 7, 9, 5, 8]})

df

sleep  job_satisfaction
0      7                 8
1      6                 5
2      8                 9
3      5                 4
4      6                 6
5      7                 8
6      6                 7
7      8                 9
8      5                 5
9      7                 8

In [4]:
correlation,_ = spearmanr(df)

In [5]:
if correlation > 0:
    interpretation = 'positive monotonic relationship between variables'
    
elif correlation <0:
    interpretation = 'negative monotonic relationship between variables'
    
else:
    interpretation = 'there is no monotonic relationshio between variables'
    
print(interpretation)

positive monotonic relationship between variables


## Question - 3
ans - 

In [16]:
import numpy as np
import pandas as pd

exercise = np.random.randint(1,18,50)
bmi = np.random.randint(20,35,50)

df = pd.DataFrame({'exercise':exercise, 'bmi':bmi})

df.head()


exercise  bmi
0         9   22
1         4   20
2        10   25
3         2   25
4         4   27

In [25]:
cor1,_ = df.corr(method = 'pearson')



In [26]:
cor2,_ = df.corr(method = 'spearman')


In [27]:
if cor1 > cor2:
    print("pearson correlation is stronger")
    
else:
    print("spearman correlation is stronger")

spearman correlation is stronger


## Question -4
ans - 

In [30]:
import numpy as np
import pandas as pd


television_hours = np.random.randint(1,10,50)
physical_level = np.random.randint(1,10,50)


df = pd.DataFrame({"television_hours":television_hours, 'physical_level':physical_level})

df.head()

television_hours  physical_level
0                 5               2
1                 6               8
2                 3               9
3                 1               6
4                 4               5

In [31]:
df.corr(method = 'pearson')

television_hours  physical_level
television_hours          1.000000        0.215104
physical_level            0.215104        1.000000

## Question - 5
ans - 

In [32]:
import numpy as np
import pandas as pd

age = [25,42,37,19,31,28]
soft_drink = ['coke', 'pepsi','mountain_dew', 'coke','pepsi','coke']


df = pd.DataFrame({"age":age, 'soft_drink':soft_drink})

df

age    soft_drink
0   25          coke
1   42         pepsi
2   37  mountain_dew
3   19          coke
4   31         pepsi
5   28          coke

In [35]:
df.corr(method = 'pearson')

/tmp/ipykernel_84/3294333281.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr(method = 'pearson')


age  encoded_soft_drink
age                 1.000000            0.769175
encoded_soft_drink  0.769175            1.000000

In [33]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()

df['encoded_soft_drink'] = encoder.fit_transform(df['soft_drink'])

In [34]:
df

age    soft_drink  encoded_soft_drink
0   25          coke                   0
1   42         pepsi                   2
2   37  mountain_dew                   1
3   19          coke                   0
4   31         pepsi                   2
5   28          coke                   0

In [38]:
df.corr( method = 'pearson')

/tmp/ipykernel_84/3887611810.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr( method = 'pearson')


age  encoded_soft_drink
age                 1.000000            0.769175
encoded_soft_drink  0.769175            1.000000

## Question - 6
ans - 

In [9]:
import numpy as np
import pandas as pd


sales_call = np.random.randint(5,15,30)
sales_made = np.random.randint(20,50,30)

df = pd.DataFrame({'sales_call':sales_call, 'sales_made':sales_made})

df.corr(method = 'pearson')
               

sales_call  sales_made
sales_call    1.000000    0.134933
sales_made    0.134933    1.000000